### ISSUE TITLE
Can you please strong name the assembly?

### ISSUE THREAD

**Issue by slavapvf** [Speedygeek/ZendeskApi_v2#479]
Could not load file or assembly 'ZendeskApi_v2, Version=3.10.7.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)   
please sign it

**Comment by mozts2005** [comment]
Strong name signing is a breaking change that adds no value. I do plan to make this change with the next major release which you can see being worked on here  https://github.com/Speedygeek/ZendeskAPI. 

For more help please see https://stackoverflow.com/questions/331520/how-to-fix-referenced-assembly-does-not-have-a-strong-name-error/10863860#10863860

**Comment by ilCosmico** [comment]
> Strong name signing is a breaking change that adds no value.

@mozts2005, Respectfully, I disagree. Implementing strong name signing is not just a breaking change; it's a crucial security and compatibility measure in the .NET Framework.

Firstly, a project not strongly named can reference an assembly that is strongly named, but the reverse is not true. This one-way compatibility is an essential consideration.

Furthermore, strong-name signing offers several significant advantages:

- Identity and Security: Strong-name signing uniquely identifies an assembly and ensures its integrity. This verification is pivotal for security, guarding against potential tampering and the insertion of malicious code.

- Interoperability: Strong-name signing is a prerequisite for placing assemblies in the Global Assembly Cache (GAC), a shared repository for use by multiple applications. Without strong-naming, an assembly cannot be installed in the GAC, limiting its usability.

- Compatibility: Strong-name signing fosters compatibility by ensuring that assemblies are loaded only by applications compiled with the corresponding strong-name key. This helps prevent compatibility issues when integrating assemblies from different sources.

**Comment by mozts2005** [comment]
Microsoft recommendation for library authors, is to not add strong naming to libraries that have already been released without strong naming. https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming. ![image](https://github.com/Speedygeek/ZendeskApi_v2/assets/626061/6127ad98-46b9-44f2-a36d-41cc107dc738)

Strong naming is not considered a security feature as is referred to in Microsoft own documentation. I do Authenticode and NuGet Package Signing as is recommended by Microsoft.  ![image](https://github.com/Speedygeek/ZendeskApi_v2/assets/626061/885f0bff-f8cb-400e-807c-1d2b42784cc6)

This is an open source project and if your corporate policy or other reasons require you to have strong name copy feel free to clone and create a copy for your organization. I would ask that you do not publish it on Nuget.org as it adds confusion for other users that do not need this feature. 

I am intending to do a complete rewrite of this library based on the system.text.json serializer a few features that I needed were released as part of .NET 8 so I have restarted this project. As this will be a new major version of the library, I will review the ability to add strong naming with the first release of this new version of the library but as this complicates the versioning of the assemblies, I need to make sure that this is properly capable and compatible with not only windows based .NET but cross platform as most of the clients that I have supported over the last few years are no longer using windows for production environments, and Strong naming has no benefits on .NET Core/5+.

TLDR: I will review strong naming options with the next major release 4.X but that will be a complete rewrite library based on system.text.JSON If I am able to add strong naming and not cause more headaches for builds, and other features like Authenticode. At this time I do not have an ETA for this release. I am hoping to do it sometime in 2024, but as this is not my day job, I cannot guarantee that.
